google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -1440,4 +1443,19 @@ public void testReadCompressedBlob() throws IOException {
     }
     blob.delete();
   }
+
+  @Test
+  public void testBlobNameNormalization() {
+    for (String blobName : BLOB_NORMALIZATION.keySet()) {
+      BlobInfo blobInfo = BlobInfo.newBuilder(BlobId.of(BUCKET, blobName))
+          .setContentType("text/plain")
+          .setContentEncoding("UTF-8")
+          .build();
+      Blob blob = storage.create(blobInfo, BLOB_NORMALIZATION.get(blobName).getBytes());
```

Comment:
This isn't necessarily proving what we want it to. In particular, without checking the code, I don't know whether `Blob.getName()` reflects the value *returned* by Storage, or whether it's just using the `BlobInfo` we provided.

I would suggest using the known-to-demonstrate-both-cases existing storage files - check you can download both of those and get the right values.